CRM/Civicase/Hook/PreProcess/ActivityFormTitleWordReplacement.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
+use CRM_Civicase_Hook_Helper_CaseTypeCategory as CaseCategoryWordReplacementHelper;
+use CRM_Utils_System as SystemUtils;
+use function ts as civicaseTs;
+
+/**
+ * Activity Form Title Word Replacement.
+ *
+ * Replaces the form title with the right translation for the activity type,
+ * depending on the selected activity's case category.
+ */
+class CRM_Civicase_Hook_PreProcess_ActivityFormTitleWordReplacement {
```

Comment:
This class share virtually same logic with https://github.com/compucorp/uk.co.compucorp.civicase/blob/349047bef3e8c79a57b5942341ddb7c9b88fde02/CRM/Civicase/Hook/PreProcess/CaseCategoryWordReplacementsForChangeCase.php except for the shouldRun function. We should just move that logic to a base class and use it from there. 
The advantages is that if something changes especially regarding the page title logic we would be able to change in one place instead of having it in two classes. We might also need the logic in another class later.

Reply:
We might not even need another class, if we can find a common way of getting the case ID from the form object. i.e if `$form->getVar('_caseId')` or something similar works for both forms. We just need to change the should run to 
```php
 $isCaseActivity = !!$form->getVar('_caseId');
return $formName == 'CRM_Case_Form_Activity' || $formName == 'CRM_Case_Form_Activity' && $isCaseActivity
```

Reply:
UPDTED: Turns out we didn't need this new class at all. The class https://github.com/compucorp/uk.co.compucorp.civicase/blob/349047bef3e8c79a57b5942341ddb7c9b88fde02/CRM/Civicase/Hook/PreProcess/CaseCategoryWordReplacementsForChangeCase.php should have catered for this already, I fixed a bug that was not making this work as expected.